### PR TITLE
[com4FlowPy] cell_counts BUG FIX and code simplification

### PR DESCRIPTION
**Pull request by ahuber-bfw**
* This PR fixes the "Cell_Counts" Output Layer of com4FlowPy --> now the value in 'cell_counts' reflects the number of release pixels that reach a certain raster-cell - the extra "track-keeping" is done in a dictionary and comes at almost no additional computational cost.
* This PR also introduces a simplification to the calculation() function in flowCore.py, so that at each initialization of a flowClass.Cell() Object is only written once, instead of multiple times, depending on the "forestBool" option.

### CODE REVIEW

File: avaframe/com4FlowPy/flowCore.py [hunk 1]
```diff
@@ -319,6 +319,7 @@ def calculation(args):
     # if not, wait for 30 secs and check again
     # should prevent the occurence of broken pipe errors or similar issues related
     # to RAM overflow
+    log = logging.getLogger(__name__)
```
Comment (fso42) [inline avaframe/com4FlowPy/flowCore.py:322]: log is never used, remove it? Or is this for a planned usage?

File: avaframe/com4FlowPy/flowCore.py [hunk 2]
```diff
@@ -370,27 +375,21 @@ def calculation(args):
...
+            startcell=True,
+            FSI=forestArray[row_idx, col_idx] if isinstance(forestArray, np.ndarray) else None,
+            forestParams=forestParams,
+        )
 
+        # dictionary of all the cells that have been processed and the number of times the cell has been visited
+        processedCells[(startcell.rowindex, startcell.colindex)] = 1
+        # helper list with indices of cells, which still need to be worked
+        # this list is extended while looping through it - the calculation for one release pixel terminates
+        # when all the cells in this list have been worked / caluclated
```
Comment (fso42) [inline avaframe/com4FlowPy/flowCore.py:392]: -> calculated (typo)

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/com4FlowPy/flowCore.py (hunk 1 above)
Comment (ahuber-bfw) [reply avaframe/com4FlowPy/flowCore.py]: used it for debugging and forgot to remove it

**Review by fso42 — APPROVED** [review]

File: avaframe/com4FlowPy/flowCore.py [hunk 3]
```diff
@@ -361,6 +364,7 @@ def calculation(args):
     startcell_idx = 0
     while startcell_idx < len(row_list):
 
+        processedCells = {}  # dictionary of cells that have been processed already
```
Comment (PaulaSp3) [inline avaframe/com4FlowPy/flowCore.py:367]: If we used a list here, I think we could change the part where we search, if the cell already exists (line 405), that is now quite inefficient. I've tried that in the [PS_flowpy_fluxSum](https://github.com/avaframe/AvaFrame/tree/PS_flowpy_fluxSum) branch.
Just an idea for the next time :)

File: avaframe/com4FlowPy/flowCore.py [hunk 4]
```diff
@@ -426,38 +423,37 @@ def calculation(args):
...
+                            startcell,
+                            FSI=forestArray[row[k], col[k]] if isinstance(forestArray, np.ndarray) else None,
+                            forestParams=forestParams,
+                                     ))
 
             z_delta_array[cell.rowindex, cell.colindex] = max(z_delta_array[cell.rowindex, cell.colindex], cell.z_delta)
             flux_array[cell.rowindex, cell.colindex] = max(flux_array[cell.rowindex, cell.colindex], cell.flux)
-            count_array[cell.rowindex, cell.colindex] += int(1)
+
+            if processedCells[(cell.rowindex, cell.colindex)] == 1:
```
Comment (PaulaSp3) [inline avaframe/com4FlowPy/flowCore.py:448]: why `` == 1`` ? 
``processedCells[(cell.rowindex, cell.colindex)]`` could also be larger, if the cell is hit more often (after lines 429, 430), so we don't consider these cells in the ``count_array``?

File: avaframe/com4FlowPy/flowCore.py (hunk 3 above)
Comment (fso42) [reply avaframe/com4FlowPy/flowCore.py:367]: @PaulaSp3 if you have an idea like this, feel free to start a new issue from here. I added a new label called flowPyDev, just tag the issue with that...

Comment (ahuber-bfw) [reply avaframe/com4FlowPy/flowCore.py:367]: I just use this dictionary as a lookup table for now, i.e. every time a cell is added to the cell_list, where the actual processing is taking place, I look if this cell (col, row) is already in the dict - if not, then the cell is added with a counter of 1, if the cell has been 'visited' already this counter is incremented.

- The lookup if an element is in a list is O(n)
- The lookup if an element is in a dictionary is O(1)

so using a list for this specific purpose would be slower imo

Comment (ahuber-bfw) [reply avaframe/com4FlowPy/flowCore.py:367]: I also tried using a dictionary for ```cell_list```, but did not manage to produce the same results as the list implementation $\rightarrow$ But we can discuss this :)

**Review by PaulaSp3 — APPROVED** [review]

**Pull request merged**